assets/training/finetune_acft_hf_nlp/src/model_converter/model_converter_adapters.py

```diff
@@ -229,20 +237,61 @@ def __init__(self, component_args: Namespace) -> None:
+        """Remove unwanted packages from conda and requirements file."""
+        if is_main_process():
+            req_file_path = os.path.join(model_save_path, "requirements.txt")
+            conda_file_path = os.path.join(model_save_path, "conda.yaml")
+            requirements = None
+            if os.path.exists(req_file_path):
+                with open(req_file_path, "r") as f:
+                    requirements = f.readlines()
+                if requirements:
+                    for package in UNWANTED_PACKAGES:
+                        requirements = [item for item in requirements if not item.startswith(package)]
+                    logger.info("Updated requirements.txt file")
+
+            conda_dict = None
+            if os.path.exists(conda_file_path):
+                with open(conda_file_path, "r") as f:
+                    conda_dict = yaml.safe_load(f)
+                if conda_dict is not None and "dependencies" in conda_dict:
+                    for i in range(len(conda_dict["dependencies"])):
+                        if "pip" in conda_dict["dependencies"][i] and isinstance(conda_dict["dependencies"][i], dict):
+                            pip_list = conda_dict["dependencies"][i]["pip"]
+                            if len(pip_list) > 0:
+                                for package in UNWANTED_PACKAGES:
+                                    pip_list = [item for item in pip_list if not item.startswith(package)]
+                                conda_dict["dependencies"][i]["pip"] = pip_list
+                                break
+                    with open(conda_file_path, "w") as f:
+                        yaml.safe_dump(conda_dict, f)
+                    logger.info("Updated conda.yaml file")
+
+    def is_t5_text_classification_finetune(self, model_type) -> bool:
+        """Check for t5 text-classification."""
+        return self.mlflow_task_type == MLFlowHFFlavourTasks.SINGLE_LABEL_CLASSIFICATION and \
+            model_type == HfModelTypes.T5
+
     def convert_model(self) -> None:
         """Convert pytorch model to oss mlflow model."""
         # load model and tokenizer
         model, tokenizer = self.load_model_and_tokenizer()
 
         self.set_mlflow_model_parameters(model)
 
-        conda_file_path = Path(self.ft_pytorch_model_path, MLFlowHFFlavourConstants.CONDA_YAML_FILE)
-        if conda_file_path.is_file():
-            self.mlflow_save_model_kwargs.update({"conda_env": str(conda_file_path)})
-            logger.info(
-                f"Found {MLFlowHFFlavourConstants.CONDA_YAML_FILE} from base model. "
-                "Using it for saving Mlflow model."
-            )
+        # Temp Fix:
+        # specific check for t5 text-classification so that base model dependencies doesn't get pass
+        # and use transformers version 4.40.0 from infer dependencies
+        if not self.is_t5_text_classification_finetune(model.config.model_type):
```

Comment:
Is this check specific to `transformers` flavor? Can we keep this agnostic to flavor?

Reply:
this check is only for t5 text classification, coz remove apex only needed for that as only for that case we move to infer dependencies.
we can't do flavor specific eg. transformers if we move it will cover lot of models, moving to infer dependencies for all oss models will not be relevent at this stage.